src/services/codefixes/convertTypedefToType.ts

```diff
@@ -48,35 +50,63 @@ registerCodeFix({
             ];
         }
     },
-    getAllCodeActions: context => codeFixAll(context, errorCodes, (changes, diag) => {
-        const node = getTokenAtPosition(diag.file, diag.start);
-        if (node) doChange(changes, node, diag.file);
-    })
+    getAllCodeActions: context => codeFixAll(
+        context,
+        errorCodes,
+        (changes, diag) => {
+            const node = getTokenAtPosition(diag.file, diag.start);
+            if (node) doChange(changes, node, diag.file, context);
+        }
+    )
 });
 
-function doChange(changes: textChanges.ChangeTracker, node: Node, sourceFile: SourceFile) {
+function doChange(
+    changes: textChanges.ChangeTracker,
+    node: Node, sourceFile: SourceFile,
+    context: textChanges.TextChangesContext
+) {
     if (isJSDocTypedefTag(node)) {
-        fixSingleTypeDef(changes, node, sourceFile);
+        fixSingleTypeDef(changes, node, sourceFile, context);
     }
 }
 
 function fixSingleTypeDef(
     changes: textChanges.ChangeTracker,
     typeDefNode: JSDocTypedefTag | undefined,
     sourceFile: SourceFile,
+    context: textChanges.TextChangesContext
 ) {
     if (!typeDefNode) return;
 
     const declaration = createDeclaration(typeDefNode);
     if (!declaration) return;
 
-    const comment = typeDefNode.parent;
+    const newLine = getNewLineOrDefaultFromHost(context.host, context.formatContext.options);
+
+    const emptyCommentBefore = isEmptyComment(

```

Comment:
I'm moderately confused; what is this empty comment detection doing? Is there a code example I've missed?

Reply:
Agree, I will try to find a better name :)

It's for cases like [this](https://github.com/microsoft/TypeScript/pull/54667/files#diff-732e152c63d6d06cf6b37f111f434a13c25b3fce62feb82666dc84e8fde0b5d0R149-R167) when a user tries to apply codefix to the @typedef at the edges or in the middle of the comment block. 

The function checks if is any content left in the comment block (text block, actually, but we know it's part of the comment) between the start and end (except `*` and `/`).

Here is an example:

```
/**
 * @typedef {string} FooString
 *
 * @typedef {number} FooNumber
 */
 ```
 
 if you try to convert `@typedef {string} FooString` to interface, the function above will check 
 ```
 /**
 * 
 ``` 
 to find out if any content is left above and if we need to close the block or just remove the rest
 
 Same for the part below:
 ```
  * @typedef {number} FooNumber
 */
 ```